### PR TITLE
XInput Support

### PR DESCRIPTION
**Pull request by admshao**
Although xcb-xinput is [enabled](https://cgit.freedesktop.org/xcb/libxcb/commit/?id=d10194a321c3db851b3ede9a98cdc95e951943aa) by default as of March, 2017 (XCB 1.12+) when compiling from source. Distros may change compiler flags and as a result of that i can't find it distributed on Debian/Ubuntu derivatives by default. Arch have xcb 1.12 with xinput support but Ubuntu 17.04+ and Debian stretch+ have xcb 1.12 without xinput support.

Users will need to install xinput packages or compile and then recompile obs.

This fixes Mantis Issue #712

### CODE REVIEW

File: libobs/obs-nix.c [hunk 1]
```diff
@@ -707,6 +714,56 @@ static inline bool fill_keycodes(struct obs_core_hotkeys *hotkeys)
...
+		xcb_connection_t *connection)
+{
+	int def_screen_idx = XDefaultScreen(context->display);
+	xcb_screen_iterator_t iter;
+
+	iter = xcb_setup_roots_iterator(xcb_get_setup(connection));
+	while (iter.rem) {
+		if (def_screen_idx-- == 0) {
+			return iter.data;
+			break;
```
Comment (RytoEX) [inline libobs/obs-nix.c]: Isn't this `break` unreachable?

Comment (admshao) [reply libobs/obs-nix.c]: Urgh.. Yeah. Unecessary. Will update in a moment

Comment (RytoEX) [reply libobs/obs-nix.c]: In that case, I think the `if` statement does not need curly braces if it only contains a single-line statement.

File: libobs/obs-nix.c [hunk 2]
```diff
@@ -633,9 +640,9 @@ static inline void add_key(obs_hotkeys_platform_t *context, obs_key_t key,
 
 	if (context->keycodes[key].list.num > 1) {
 		blog(LOG_DEBUG, "found alternate keycode %d for %s "
-		                "which already has keycode %d",
-		                code, obs_key_to_name(key),
-		                (int)context->keycodes[key].list.array[0]);
+				"which already has keycode %d",
+				code, obs_key_to_name(key),
+				(int)context->keycodes[key].list.array[0]);
```
Comment (jp9000) [inline libobs/obs-nix.c]: This whitespace change is not relevant to the commit nor needed and should be removed.

File: libobs/obs-nix.c [hunk 3]
```diff
@@ -813,7 +978,7 @@ static bool key_pressed(xcb_connection_t *connection,
 
 	} else if (key == OBS_KEY_META) {
 		pressed = keycode_pressed(reply, context->super_l_code) ||
-		          keycode_pressed(reply, context->super_r_code);
+			  keycode_pressed(reply, context->super_r_code);
```
Comment (jp9000) [inline libobs/obs-nix.c]: This appears to be another whitespace change that wasn't necessary/relevant.  Make sure to review your diffs in the future to make sure the code you've committed is relevant to the changes you're trying to submit.

**Pull request merged**